python/lsst/cell_coadds/_fits.py

```diff
@@ -0,0 +1,407 @@
+            The data from the FITS file.
+        header : `dict`
+            The header of the FITS file.
+        common : `CommonComponents`
+            The common components of the coadd.
+
+        Returns
+        -------
+        coadd : `SingleCellCoadd`
+            The coadd read from the file.
+        """
+        buffer_x, buffer_y = (header["OCELL1"] - header["GRCELL1"]) // 2, (
+            header["OCELL2"] - header["GRCELL2"]
+        ) // 2
+        psf = ImageD(
+            array=data["psf"].astype(np.float64),
+            xy0=Point2I(-(header["PSFSIZE1"] // 2), -(header["PSFSIZE2"] // 2)),
+        )  # use the variable
+        xy0 = Point2I(
+            header["GRCELL1"] * data["cell_id"][0] - buffer_x,
+            header["GRCELL2"] * data["cell_id"][1] - buffer_y,
+        )
+        mask = afwImage.Mask(data["mask"].astype(np.int32), xy0=xy0)
+        image_planes = OwnedImagePlanes(
+            image=ImageF(
+                data["image"].astype(np.float32),
+                xy0=xy0,
+            ),
+            mask=mask,
+            variance=ImageF(data["variance"].astype(np.float32), xy0=xy0),
+            noise_realizations=[],
+            mask_fractions={},
+        )
+
+        identifiers = CellIdentifiers(
+            cell=Index2D(data["cell_id"][0], data["cell_id"][1]),
+            skymap=common.identifiers.skymap,
+            tract=common.identifiers.tract,
+            patch=common.identifiers.patch,
+            band=common.identifiers.band,
+        )
+
+        return SingleCellCoadd(
+            outer=image_planes,
+            psf=psf,
+            inner_bbox=Box2I(
+                minimum=Point2I(
+                    header["GRCELL1"] * data["cell_id"][0],
+                    header["GRCELL2"] * data["cell_id"][1],
+                ),
+                maximum=Point2I(
+                    header["GRCELL1"] * (data["cell_id"][0] + 1) - 1,
+                    header["GRCELL2"] * (data["cell_id"][1] + 1) - 1,
+                ),
+            ),
+            common=common,
+            identifiers=identifiers,
+            inputs=None,  # type: ignore[arg-type]  # TODO: Pass a sensible value here.
+        )
+
+    # TODO: Make this reader a context manager that handles file closures.
+
+    def readBBox(self) -> Box2I:
+        raise NotImplementedError("Not yet implemented for MultipleCellCoadd")
+
+    def readPsf(self) -> ImageF:
+        raise NotImplementedError("Not yet implemented for MultipleCellCoadd")
+
+    def readFilter(self) -> str:
+        """Read the filter information from the FITS file.
+
+        Returns
+        -------
+        band : `str`
+            The filter information read from the FITS file.
+        """
+        with fits.open(self.filename) as hdu_list:
+            band = hdu_list[1].header["BAND"]
+        return band
+
+    def readXY0(self) -> Point2I:
+        raise NotImplementedError("Not yet implemented for MultipleCellCoadd")
```

Comment:
Let's just delete all of the `read<something>` methods other than `readWcs`:
- `readBBox` and `readXY0` aren't explicit about whether they refer to the inner or outer bbox, and the user would be better off getting either from a `BaseSkyMap` object anyway;
- `readPsf` might be useful, but it might need to take arguments or return something other than `ImageF`, so we're not just ready to add a placeholder yet.
- `readFilter` should probably return an `lsst.afw.image.Filter` object rather than a `str` if we really wanted to keep it, but I don't actually think the ability to read the filter label from an `Exposure` object is at all useful - it's just there historically and hence at least a bit of a pain to remove - and I don't think it'll be useful here, either.

Reply:
Removed all the `read<something>` methods other than `readWcs`.